File: test/prime.py
# prime.py
from sys import argv
from math import sqrt

def isprime(n, primes):
    nsqrt = sqrt(n)
    for x in primes:
        if nsqrt < x: break
        if n % x == 0: return False
    return True

primes = []
for x in xrange(2, int(argv[1])):
    if isprime(x, primes): primes.append(x)
print primes[-1]

File: test/prime.cpp
// prime.cpp
#include <math.h>
#include <stdlib.h>
#include <stdio.h>
#include <vector>

int isprime(int n, const std::vector<int>& primes) {
    const double nsqrt = sqrt(n);
    for(const auto& curs : primes) {
        if(nsqrt < curs) { break; }
        if(n % curs == 0) { return 0; }
    }
    return n;
}

int main(int argc, const char** argv) {
    const int primes_capacity = atoi(argv[1]);
    std::vector<int> primes;
    primes.reserve(primes_capacity);
    for(int Li = 2; Li < primes_capacity; ++Li) {
        if(int prime = isprime(Li, primes)) {
            primes.push_back(prime);
        }
    }
    printf("%d\n", primes.back());
    return 0;
}

